packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/data_type.rs

```diff
@@ -32,28 +34,32 @@ impl<C: AsClient> PostgresStore<C> {
         let dependency_status = dependency_context
             .ontology_dependency_map
             .insert(data_type_id, current_resolve_depth);
-        let data_type: Option<&OntologyVertex> = match dependency_status {
-            DependencyStatus::Unknown => {
-                if let Some(data_type) = subgraph.vertices.ontology.get(data_type_id) {
-                    Some(data_type)
-                } else {
-                    let data_type = Read::<DataTypeWithMetadata>::read_one(
-                        self,
-                        &Filter::<DataType>::for_ontology_type_edition_id(data_type_id),
-                    )
-                    .await?;
-                    Some(
-                        subgraph
-                            .vertices
-                            .ontology
-                            .entry(data_type_id.clone())
-                            .or_insert(OntologyVertex::DataType(Box::new(data_type))),
-                    )
+        let data_type = match dependency_status {
+            DependencyStatus::Unresolved => {
+                match subgraph
+                    .vertices
+                    .ontology
+                    .raw_entry_mut()
+                    .from_key(data_type_id)
+                {
+                    RawEntryMut::Occupied(entry) => Some(entry.into_mut()),
```

Comment:
Same comment here about mutable refs

Reply:
Wish we could do it with `from_key().or_insert_with` but the `?` makes that impossible it seems

Reply:
`or_insert_with` is mainly not working because of the `async` nature of `Read`